apps/advisor/pkg/app/utils.go

```diff
@@ -72,14 +72,32 @@ func processCheck(ctx context.Context, client resource.Client, obj resource.Obje
 		UserUID:      uid,
 		FallbackType: typ,
 	})
-	// Run the checks
-	report, err := check.Run(ctx, &c.Spec)
+	// Init the checks
+	err = check.Init(ctx)
```

Comment:
Just want to confirm that the expectation is that we will always call `Init` before calling `Run`? Is there any place other than here where the checks get executed?

Reply:
> Just want to confirm that the expectation is that we will always call Init before calling Run?

Yes, I can write that in the function documentation. `Init` should be called before running the `Steps` or calling `ItemsLen`. 
We could do the initialization in the `Steps` function but in any case, `Steps` (and the initialization) should be run before calling `ItemsLen`, which could be a little bit less intuitive. That's why I left the `Init` function.

> Is there any place other than here where the checks get executed?

Nope, the checks are only executed in this function (apart from tests).